### PR TITLE
refactor(jellyfish-transaction): move vault dftx into its own file

### PR DESCRIPTION
**Pull request by cwkang1998**
What this PR does / why we need it:

Part of #811 and #1145, this specifically seperates loan and vault in the jellyfish-transaction package.

Which issue(s) does this PR fixes?:

Fixes #
Additional comments?:

### CODE REVIEW

**Comment by cwkang1998** [comment]
Since there is not much change in the tests except for the imports, and the exported members are done through index.ts, I think it should be okay not to have duplicated test cases here to cater for old changes as per comments in https://github.com/DeFiCh/jellyfish/pull/1251#issue-1177856150.

If its still neccessary please by all means let me know, and I'll add them in.

File: packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts [hunk 1]
```diff
@@ -169,108 +169,6 @@ export class CUpdateLoanToken extends ComposableBuffer<UpdateLoanToken> {
   }
 }
 
-/**
- * CreateVault DeFi Transaction
- */
-export interface CreateVault {
```
Comment (ivan-zynesis) [inline packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts:175]: Export `vault` dftx here. Downstream should not break.

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]

**Review by ivan-zynesis — DISMISSED** [review]

File: packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts [hunk 2]
```diff
@@ -367,54 +265,4 @@ export class CPaybackLoanV2 extends ComposableBuffer<PaybackLoanV2> {
...
-  composers (pab: PlaceAuctionBid): BufferComposer[] {
-    return [
-      ComposableBuffer.hexBEBufferLE(32, () => pab.vaultId, v => pab.vaultId = v),
-      ComposableBuffer.uInt32(() => pab.index, v => pab.index = v),
-      ComposableBuffer.single<Script>(() => pab.from, v => pab.from = v, v => new CScript(v)),
-      ComposableBuffer.single<TokenBalanceVarInt>(() => pab.tokenAmount, v => pab.tokenAmount = v, v => new CTokenBalanceVarInt(v))
-    ]
-  }
-}
+export * from './dftx_vault'
```
Comment (ivan-zynesis) [inline packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts:268]: Add comment label them as deprecated

File: packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { BufferComposer, ComposableBuffer } from '@defichain/jellyfish-buffer'
-import { CTokenBalance, CTokenBalanceVarInt, TokenBalance, TokenBalanceVarInt } from './dftx_balance'
+import { CTokenBalance, TokenBalance } from './dftx_balance'
```
Comment (canonbrother) [inline packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts:2]: here i think `dftx_loans` need support the removed vault function as well
try export import here will do

Comment (cwkang1998) [reply packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts:2]: I already exported vaults functions via https://github.com/DeFiCh/jellyfish/pull/1252/files#diff-2f82061c33af424e1be7b95e01271a0a9c3351225da34587cb8db8ca3764c604R271
 Its a `*` export so should be exporting everything from vault as well.

**Review by canonbrother — APPROVED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Review by jingyi2811 — APPROVED** [review]

File: packages/jellyfish-transaction/__tests__/script/dftx/dftx_vault/WithdrawFromVault.test.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+} from '../../../../src/script/dftx/dftx_vault'
+import { OP_CODES } from '../../../../src/script'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import { OP_DEFI_TX } from '../../../../src/script/dftx'
+import BigNumber from 'bignumber.js'
+
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
+    /**
+     * WithdrawFromVault : {
```
Comment (jingyi2811) [inline packages/jellyfish-transaction/__tests__/script/dftx/dftx_vault/WithdrawFromVault.test.ts:14]: Just for knowledge sharing, we need to add at least 3 fixtures for `should bi-directional buffer-object-buffer`
At you are copied this from the previous programmer, we can ignore this for now.

**Review by surangap — APPROVED** [review]

**Pull request merged**